vsintegration/src/FSharp.Editor/CodeFix/AddOpenCodeFixProvider.fs

```diff
@@ -148,7 +148,7 @@ type internal FSharpAddOpenCodeFixProvider
         asyncMaybe {
             let! options = projectInfoManager.TryGetOptionsForEditingDocumentOrProject context.Document
             let! sourceText = context.Document.GetTextAsync(context.CancellationToken)
-            let! parsedInput, checkResults = checker.ParseAndCheckDocument(context.Document, options, sourceText)
+            let! _, parsedInput, checkResults = checker.ParseAndCheckDocument(context.Document, options, sourceText)
```

Comment:
Is there any chance that the Async completion of TypeCheckOneFile may now cause multiple overlapping ParseAndCheckDocument on the exact same document, where the cached results of one are not provided to the next? e.g.

1. Start Request A 
2. Start Request B 
3. Look up cache for request A 
4. Lookup cahce for Request B 
5. Complete A and write cache entry
6. Complete Request B and write cache entry

I don't think so because the reactor thread still serializes things?  But I just want to check, since we are very much relying on the FCS cahces to amortize the cost of multiple ParseAndCheckFileInProject calls over the exact same file and source text.  (We should make sure this is under test too - I think it is through the parse/check counters)